docs/reference/rollup/apis/rollup-api.asciidoc

```diff
@@ -0,0 +1,191 @@
+For example, you can roll up hourly data into daily or weekly summaries.
+
+[source,console]
+----
+POST /my-index-000001/_rollup
+{
+  "rollup_index": "my-rollup-index",
+  "groups": {
+    "date_histogram": {
+      "field": "@timestamp",
+      "calendar_interval": "1d"
+    },
+    "terms": {
+      "fields": [
+        "my-keyword-field",
+        "my-other-keyword-field"
+      ]
+    }
+  },
+  "metrics": [
+    {
+      "field": "my-numeric-field",
+      "metrics": [
+        "min",
+        "max",
+        "avg"
+      ]
+    }
+  ]
+}
+----
+// TEST[setup:my_index]
+
+[[rollup-api-request]]
+==== {api-request-title}
+
+`PUT /<index>/_rollup`
+
+[[rollup-api-prereqs]]
+==== {api-prereq-title}
+
+If the {es} {security-features} are enabled, you must have the `manage` cluster
+privilege to use this API. See <<security-privileges>>.
+
+[[rollup-api-path-params]]
+==== {api-path-parms-title}
+
+`<index>`::
+(Required, string)
+Index to roll up. Does not support <<multi-index,multi-target syntax>> or
+wildcards (`*`).
++
+You can't use this argument to directly target a <<data-streams,data stream>> or
+<<indices-aliases,index alias>>. However, you can target their backing or
+concrete indices.
+
+
+[role="child_attributes"]
+[[rollup-api-request-body]]
+==== {api-request-body-title}
+
+`rollup_index`::
+(Required, index)
+Index that stores the rollup results. Cannot be a <<data-streams,data stream>>
+or <<indices-aliases,index alias>>.
++
+If this index does not exist, the request creates it. If the `<index>` is a
+backing index for a data stream, this new index is a backing index of the same
+stream.
+
+`groups`::
+(Required, object)
+Aggregates and stores fields in the rollup.
++
+.Properties of `groups`
+[%collapsible%open]
+====
+`date_histogram`::
+(Required,
+<<search-aggregations-bucket-datehistogram-aggregation,`date_histogram`
+aggregation>> object)
+Groups documents in the original `<index>` based on a provided time interval.
++
+.Properties of `date_histogram`
+[%collapsible%open]
+=====
+`field`::
+(Required, string)
+<<date,`date`>> or <<date_nanos,`date_nanos`>> field containing the `<index>`'s
+timestamp. If you're rolling up a backing index or using the {ecs-ref}[Elastic
+Common Schema (ECS)], we recommend using `@timestamp`.
+
+`calendar_interval` or `fixed_interval`::
+(Required, <<time-units,time units>>)
+Time interval used to group documents. For differences between
+`calendar_interval` and `fixed_interval`, see <<calendar_and_fixed_intervals>>.
++
+TIP: Choose this value carefully. You won't be able to use a smaller interval
+later. For example, you can't aggregate daily rollups into hourly
+summaries. However, smaller time intervals can greatly increase the size of your
+`rollup_index`.
+
+`time_zone`::
+(Optional, string)
+Time zone for the `field`. Valid values are ISO 8601 UTC offsets, such as
+`+01:00` or `-08:00`, and IANA time zone IDs, such as `America/Los_Angeles`.
+Defaults to `+00:00` (UTC).
+=====
```

Comment:
`delay` is also a valid param for `date_histogram`, but I chose not to include it as it seems to be no-op for the API endpoint. Let me know if that's not the case.

Reply:
This is one of those WIP leaks. the config is being re-used, but you're right, the `delay` option is not intended to be used in this action, so it should not be mentioned